enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+- Reconfiguration of the router.
+
+## Motivation
+
+### User Stories
+
+- As an administrator, I want to provision EIPs and use them with an NLB IngressController.
+- As a user or admin, I want to ensure EIPs are used with NLB on default router at install time.
+- As a user, I want to reconfigure default router to use EIPs.
+- As a user of OpenShift on AWS (ROSA), I want to use static IPs (and therefore AWS Elastic IPs) so that 
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+
+### Goals
+- Users are able to use EIP for a NLB Ingress Controller.
+- Users are able to create a NLB default Ingress Controller during install time with the EIPs specified during install.
+- Any existing ingress controller should be able to be reconfigured to use EIPs. 
+
+### Non-Goals
+
+- Creation of EIPs in AWS.
+- Keep a track of changes in the EIP in the AWS environment.
+- Static IP usage with NLBs for API server.
+
+## Proposal
+
+In this enhancement we would be adding API fields in the installer and the ingress controller CRD
+to set the AWS EIP for the NLB Ingress Controller. The cluster ingress operator will then create 
+the ingress controller and get the EIP assigned to the service LoadBalancer with the annotation 
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+#### Set EIP using an installer:
+1. Cluster administrator creates adds the following in the install-config.yaml
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: NLB
+      networkLoadBalancer:
+        eip-allocations: eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted> 
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```  
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ # Please edit the object below. Lines beginning with a '#' will be ignored,
+ # and an empty file will abort the edit. If an error occurs while saving this file will be
+ # reopened with the relevant failures.
+ #
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadBalancer:
+     platform:
+       aws:
+         type: NLB
+           networkLoadBalancer:
+             eip-allocations: eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>
+       type: AWS
```

Comment:
What happens to the subnets added in the installer config?

Reply:
Addition of subnets is a different EP. However the eips should match the number of `public subnets`.

Reply:
@miheer I don't think it is acceptable for that to be a user responsibility.  We will need to add some validations to make sure the number of public subnets matches the number of EIPs.  cc @Miciah for feedback.

Reply:
We need to distinguish between these cases:

- The cluster-admin specified invalid EIPs in `install-config.yaml` at installation time (day 0).
- The cluster-admin specified invalid EIPs in `ingresses.config.openshift.io/cluster` or an IngressController CR (day 2).

It seems to me that the installer can feasibly validate the install config for day 0 (but better verify that with the Installer team).  However, it is impractical to add API validation for day 2 as validating `eipAllocations` requires listing EIPs using the AWS API, which is not something we can do with CEL (it's not something we would typically do even with an admission plugin or webhook).  In my opinion, given the constraints and the fact that this the workflow already requires active participation by the cluster-admin, it is reasonable to put the burden on the cluster-admin to verify that the EIPs match up with public subnets, as long as we **clearly document** this expectation.  If there is a mismatch, then cloud-provider-aws and cloud-controller-manager's service manager will report an error, and cluster-ingress-operator should reflect this error to the IngressController CR's status.

Reply:
second point is already covered in Open Questions

Reply:
first point is added in the `Open Questions` as I need to verify this with the installer team.